src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/SuggestedAction.cs

```diff
@@ -179,10 +166,12 @@ protected virtual void InnerInvoke(IProgressTracker progressTracker, Cancellatio
                 using (Logger.LogBlock(
                     FunctionId.CodeFixes_ApplyChanges, KeyValueLogMessage.Create(LogType.UserAction, m => CreateLogProperties(m)), cancellationToken))
                 {
-                    // Note: we want to block the UI thread here so the user cannot modify anything while the codefix applies
-                    _isApplied = EditHandler.Apply(Workspace, getFromDocument(),
+                    var document = this.SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
```

Comment:
we had all this logic to pass in a lambda here to just be called here to fetch the doc... so that all got removed and we simply fetch the doc here.